db/migrate/20210627154945_create_bamposts.rb

```diff
@@ -0,0 +1,10 @@
+class CreateBamposts < ActiveRecord::Migration[6.1]
+  def change
+    create_table :bamposts do |t|
+      t.text :content
+      t.integer :user_id
```

Comment:
You need to relate your Users and Bamposts table, by adding foreign_key constraint to user_id column of your Bamposts table.  Short explanation: type the following to your command line: 
`rails generate migration add_user_reference_to_bamposts user:references`. 
Then type: 
`rails db:migrate`.

Long explanation: 

Normally you would add this constraint when adding user_id column to bamposts table for the first time (not to an alrady existing column as in your case) - using `rails g migration AddUserToBamposts user:references` (user is the column to be added, you say user instead of user_id - rails convention 😕 )  - this command will create the following migration file for you: 

```
class AddUserToBamposts < ActiveRecord::Migration
  def change
    add_reference :bamposts, :user, index: true
    add_foreign_key :bamposts, :users
  end
end
```

The above command basically does 3 things: 

1) adds user_id column to bamposts table 
2) adds user_id as a foreign key - so now bamposts will extract user_id info from the id column of users table 
3) adds an index on user_id column so it is easier to look for - for when we need to find a post based on user_id/when we want to find the owner of the post.

You have already added user_id column to bamposts table, but you have not added foreign key constraint yet. You have also associated these tables together using: belongs_to and has_many, but we need a foreign key reference to 'relate' these tables. To add a foreign key constraint to an already existing column (user_id), you can type this in your command line:

`rails generate migration add_user_reference_to_bamposts user:references`. This will create the following migration file: 

```class AddUserReferenceToBamposts < ActiveRecord::Migration[5.1]
  def change
    add_reference :bamposts, :user, foreign_key: true
  end
end
```

Don't forget to run `rails db:migrate` after creating your migrations. You can then associate these tables using "belong_to" and "has_many", which you already have done.  Let me know if you need me to explain any of this - also you can find a good explanation [here](https://stackoverflow.com/questions/22815009/add-a-reference-column-migration-in-rails-4):